avocado/utils/iso9660.py

```diff
@@ -293,11 +332,19 @@ def close(self):
 
         :rtype: None
         """
-        if os.path.ismount(self.mnt_dir):
-            process.run('fuser -k %s' % self.mnt_dir, ignore_status=True)
-            process.run('umount %s' % self.mnt_dir)
-
-        shutil.rmtree(self.mnt_dir)
+        if self._mnt_dir:
+            if os.path.ismount(self._mnt_dir):
+                process.run('fuser -k %s' % self.mnt_dir, ignore_status=True)
+                process.run('umount %s' % self.mnt_dir)
```

Comment:
Since this `Iso9660Mount` is intended to be used only when user `can_sudo()` this would be fine, but, notice that `can_sudo()` can return true if uid is 0 (which would make it work here), but would fail if it's selected because running `process.system_output("id -u", sudo=True, ...)` would return true.  To be on the safe side, `sudo` would have to be used here also.

Either that, or the current check for selecting this module (`can_mount()`) would have to be changed.